org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/core/compiler/IProblem.java

```diff
@@ -2564,4 +2564,10 @@ public interface IProblem {
 	 * @since 3.35
 	 */
 	int SyntheticAccessorNotEnclosingMethod = MethodRelated + 1990;
+
+	/**
+	 * @since 3.36
+	 * @noreference preview feature
+	 */
+	int FieldNameCannotBeUnderscore = PreviewRelated + 2000;
```

Comment:
This program:

```
public class X {
	int _;
	
	X(int _) {
		try {
			
		} catch (NullPointerException _) {
			
		}
	}
	
}
```

triggers two errors from javac but only one from ECJ - method/ctor parameters may not be unnamed. Please also check if any of the existing IProblem constants may be reused. 

Javac error message is much more informational - we may want to emulate

Reply:
I adjusted the new error message I added to be similar to the javac one, and use this new error message consistently when `_` is used in the wrong spot when using Java 21 with preview enabled.